index.js

```diff
@@ -1,88 +1,121 @@
 // name = "Not Raffy" 
 // => TypeError
-// console.log("%c----------", "color: red") 
-
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 3 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red")
+
+const drinkWater = currentThirstLevel => {
+  console.log("Man I sure am thirsty");
+  currentThirstLevel -= 1;
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel;
+}
+
 
-// console.log(drinkWater(12))
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+const sameSameButDifferent = (arg1, arg2) => {
+  if (arg1 === arg2){
+    return "same same"
+  } else if (arg1 == arg2){
+    return "same same (but different)"
+  } else {
+    return "different"
+  }
+}
 
-// console.log(sameSameButDifferent(5, 5)) 
+console.log(sameSameButDifferent(5, 5)) 
 // // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+console.log(sameSameButDifferent(123, "123")) 
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+console.log(sameSameButDifferent(5, 7)) 
 // // => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, "122")) 
 // // => "different"
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 5 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+console.log("%cQuestion 5", "color: red")
+const student1 = { name: "Duane", grade: 88 }
+
+const updateGrade = (student, newGrade) => {
+  student.grade = newGrade;
+  return student;
+}
+
+updateGrade(student1, 92)
+console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%c----------", "color: red")
 
 
 
 // ***** Question 6 *****
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+console.log("%cQuestion 6", "color: red")
+
+const users = [ 
+  { 
+    name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+  },
+  { 
+    name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+  }
+]
+
+const printNameAndPhones = obj => {
+  for (const key in obj){
+    console.log(obj[key]);
+  }
+}
+
```

Comment:
Nice! This works, but since this deliverable was specifically asking to print the users name and phone numbers, I would be a little more explicit in calling those specific attributes in your logs. And because `users` is an array, it might actually make more sense to use a `forEach` or `forOf` to iterate over the array. Your `for in` works just fine, but is typically reserved for objects. Alternate implementations could look something like this:

```javascript
 for (let user of users) {
  console.log(user.name);
  console.log(`Cell: ${user.phones.cell}`);
  console.log(`Office: ${user.phones.office}`);
 }
```
- Here's how it would look with forEach:
```javascript
 users.forEach(function(user) {
  console.log(user.name);
  console.log(`Cell: ${user.phones.cell}`);
  console.log(`Office: ${user.phones.office}`);
 })
```